experimental_habitat_implementation.py

```
@@ -144,7 +144,13 @@ def spawn_experiment(self, experiment: ExperimentalSystem, containment_rules: Di
         experiment.resource_limits = resource_limits
         
         # Create isolated workspace
-        exp_dir = os.path.join(self.temp_dir, experiment.name)
+        exp_dir = os.path.abspath(os.path.join(self.temp_dir, experiment.name))
+
+        # Security check: Prevent path traversal
+        temp_dir_abs = os.path.abspath(self.temp_dir)
+        if os.path.commonpath([temp_dir_abs]) != os.path.commonpath([temp_dir_abs, exp_dir]):
+            raise ValueError(f"Security violation: Experiment name '{experiment.name}' attempts path traversal.")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)